tests/utils/mock_server.py

```diff
@@ -2454,4 +2479,5 @@ def _debug(self):
     if mockserver_bind:
         kwargs["host"] = mockserver_bind
 
+    socket.socket = mock_socket_socket
```

Comment:
really surprised flask doesn't have a built-in mechanism for such a situation. hacky hack! 😍

Reply:
yep, i think it is because they have a variety of http backends.  would have been nice though.   The REUSEADDR would have worked too, but this feels more robust to me.